### PR TITLE
[API][AUTO] Fail to get PERF_COUNT from compiled_model

### PR DESCRIPTION
**Pull request by panhaiqi**
### Details:
 - *Fail to get PERF_COUNT from compiled_model*

### Tickets:
 - *CVS-130349*

### CODE REVIEW

File: src/plugins/auto/src/auto_compiled_model.cpp [hunk 1]
```diff
@@ -48,21 +48,15 @@ ov::Any AutoCompiledModel::get_property(const std::string& name) const {
                                                     ov::device::priorities,
                                                     ov::device::properties,
                                                     ov::hint::model_priority,
-                                                    ov::loaded_from_cache};
+                                                    ov::loaded_from_cache,
+                                                    ov::enable_profiling};
         return ro_properties;
```
Comment (songbell) [inline src/plugins/auto/src/auto_compiled_model.cpp:53]: -->RW

File: src/plugins/auto/src/auto_compiled_model.cpp [hunk 2]
```diff
@@ -48,21 +48,15 @@ ov::Any AutoCompiledModel::get_property(const std::string& name) const {
...
     };
-    const auto& to_string_vector = [](const std::vector<ov::PropertyName>& properties) {
-        std::vector<std::string> ret;
-        for (const auto& property : properties) {
-            ret.emplace_back(property);
-        }
-        return ret;
-    };
-    if (name == ov::supported_properties) {
+    if (name == ov::supported_properties || name == ov::enable_profiling) {
```
Comment (songbell) [inline src/plugins/auto/src/auto_compiled_model.cpp:59]: should not be added here?
also please enable same for auto ctput and add some tests

**Review by songbell — CHANGES_REQUESTED** [review]

File: src/plugins/auto/src/auto_compiled_model.cpp (hunk 2 above)
Comment (riverlijunjie) [reply src/plugins/auto/src/auto_compiled_model.cpp:59]: @panhaiqi  please add a new item for ov::enable_profiling.

File: src/plugins/auto/src/auto_compiled_model.cpp [hunk 3]
```diff
@@ -51,7 +51,7 @@ ov::Any AutoCompiledModel::get_property(const std::string& name) const {
         return ro_properties;
     };
     const auto& default_rw_properties = []() {
-        std::vector<ov::PropertyName> rw_properties{};
+        std::vector<ov::PropertyName> rw_properties{ov::enable_profiling};
```
Comment (peterchen-intel) [inline src/plugins/auto/src/auto_compiled_model.cpp:54]: This should be RO in compiledmodel

File: src/plugins/auto/src/auto_compiled_model.cpp [hunk 4]
```diff
@@ -63,6 +63,12 @@ ov::Any AutoCompiledModel::get_property(const std::string& name) const {
         supported_properties.insert(supported_properties.end(), ro_properties.begin(), ro_properties.end());
         supported_properties.insert(supported_properties.end(), rw_properties.begin(), rw_properties.end());
         return decltype(ov::supported_properties)::value_type(supported_properties);
+    } else if (name == ov::enable_profiling) {
+        auto rw_properties = default_rw_properties();
```
Comment (peterchen-intel) [inline src/plugins/auto/src/auto_compiled_model.cpp:67]: m_need_perf_counters in m_context is the variable which store the value of ov::enable_profiling property.

File: src/tests/functional/plugin/shared/src/behavior/compiled_model/properties.cpp [hunk 5]
```diff
@@ -99,6 +99,11 @@ TEST_P(OVClassCompiledModelPropertiesTests, canCompileModelWithPropertiesAndChec
     }
 }
 
+TEST_P(OVClassAutoCheckProfiling, canCompileModelWithProfilingAndCheckProfiling) {
+    auto compiled_model = core->compile_model(model, target_device, properties);
+    OV_ASSERT_NO_THROW(auto enable_profiling = compiled_model.get_property(ov::enable_profiling));
+}
+
```
Comment (songbell) [inline src/tests/functional/plugin/shared/src/behavior/compiled_model/properties.cpp:106]: we should avoid enable Auto specific cases in shared tests
and try leverage existing test or use unit tests to cover

File: src/tests/functional/plugin/shared/include/behavior/compiled_model/properties.hpp [hunk 6]
```diff
@@ -58,6 +58,7 @@ using OVClassCompiledModelPropertiesDefaultTests = OVClassCompiledModelPropertie
 using OVClassCompiledModelPropertiesIncorrectTests = OVClassCompiledModelPropertiesTests;
 using OVCompiledModelIncorrectDevice = OVClassBaseTestP;
 using OVClassCompileModelWithCorrectPropertiesTest = OVClassCompiledModelPropertiesTests;
+using OVClassAutoCheckProfiling = OVClassCompiledModelPropertiesTests;
```
Comment (songbell) [inline src/tests/functional/plugin/shared/include/behavior/compiled_model/properties.hpp:61]: should avoid this

File: src/tests/functional/plugin/shared/src/behavior/compiled_model/properties.cpp (hunk 5 above)
Comment (yangwang201911) [reply src/tests/functional/plugin/shared/src/behavior/compiled_model/properties.cpp:106]: please refer to the AUTO test case [here](https://github.com/openvinotoolkit/openvino/blob/f55e967dd5a65d2f7f48f8e1197f20f824b7fe28/src/plugins/auto/tests/functional/shared_tests_instances/behavior/ov_executable_network/properties.cpp#L61).

**Comment by yangwang201911** [comment]
also should consider to enable `ov::enable_profiling` in below test suits for AUTO/MULTI
https://github.com/openvinotoolkit/openvino/blob/f8d071002b6fe0bca2617558221850e01fe202ad/src/plugins/auto/tests/functional/shared_tests_instances/behavior/ov_plugin/properties_tests.cpp#L75
https://github.com/openvinotoolkit/openvino/blob/f8d071002b6fe0bca2617558221850e01fe202ad/src/plugins/auto/tests/functional/shared_tests_instances/behavior/ov_plugin/properties_tests.cpp#L46

File: src/plugins/auto/src/cumulative_compiled_model.cpp [hunk 7]
```diff
@@ -47,7 +47,8 @@ ov::Any AutoCumuCompiledModel::get_property(const std::string& name) const {
                                                     ov::device::properties,
                                                     ov::hint::model_priority,
                                                     ov::loaded_from_cache,
-                                                    ov::intel_auto::schedule_policy};
+                                                    ov::intel_auto::schedule_policy,
+                                                    ov::enable_profiling};
```
Comment (songbell) [inline src/plugins/auto/src/cumulative_compiled_model.cpp:51]: we can optimize by move common code path to base class, @yangwang201911 FYI, can be in another PR

**Review by songbell — APPROVED** [review]

**Review by yangwang201911 — APPROVED** [review]

File: src/plugins/auto/src/cumulative_compiled_model.cpp (hunk 7 above)
Comment (yangwang201911) [reply src/plugins/auto/src/cumulative_compiled_model.cpp:51]: got it

**Review by yangwang201911 — APPROVED** [review]

**Review by wangleis — APPROVED** [review]
